lib/adbc_connection.ex

```diff
@@ -55,9 +55,158 @@ defmodule Adbc.Connection do
     end
   end
 
+  @doc """
+  Get a string type option of the connection.
+  """
+  @spec get_string_option(pid(), atom() | String.t()) :: {:ok, String.t()} | {:error, String.t()}
+  def get_string_option(conn, key) when is_pid(conn) do
+    case GenServer.call(conn, {:get_option, to_string(key)}) do
+      {:ok, value} ->
+        {:ok, value}
+
+      {:error, reason} ->
+        {:error, error_to_exception(reason)}
+    end
+  end
+
+  @doc """
+  Get a binary (bytes) type option of the connection.
+  """
+  @spec get_binary_option(pid(), atom() | String.t()) :: {:ok, binary()} | {:error, String.t()}
+  def get_binary_option(conn, key) when is_pid(conn) do
+    case GenServer.call(conn, {:get_option, :bytes, to_string(key)}) do
+      {:ok, value} ->
+        {:ok, value}
+
+      {:error, reason} ->
+        {:error, error_to_exception(reason)}
+    end
+  end
+
+  @doc """
+  Get an integer type option of the connection.
+  """
+  @spec get_integer_option(pid(), atom() | String.t()) :: {:ok, integer()} | {:error, String.t()}
+  def get_integer_option(conn, key) when is_pid(conn) do
+    case GenServer.call(conn, {:get_option, :int, to_string(key)}) do
+      {:ok, value} ->
+        {:ok, value}
+
+      {:error, reason} ->
+        {:error, error_to_exception(reason)}
+    end
+  end
+
+  @doc """
+  Get a float type option of the connection.
+  """
+  @spec get_float_option(pid(), atom() | String.t()) :: {:ok, float()} | {:error, String.t()}
+  def get_float_option(conn, key) when is_pid(conn) do
+    case GenServer.call(conn, {:get_option, :double, to_string(key)}) do
+      {:ok, value} ->
+        {:ok, value}
+
+      {:error, reason} ->
+        {:error, error_to_exception(reason)}
+    end
+  end
+
+  @doc """
+  Set option for the connection.
+
+  - If `value` is an atom or a string, then corresponding string option will be set.
+  - If `value` is a `{:byte, binary()}`-tuple, then corresponding binary option will be set.
+  - If `value` is an integer, then corresponding integer option will be set.
+  - If `value` is a float, then corresponding float option will be set.
+  """
+  @spec set_option(
+          pid(),
+          atom() | String.t(),
+          atom() | {:bytes, binary()} | String.t() | number()
+        ) ::
+          :ok | {:error, String.t()}
+  def set_option(conn, key, value) when is_pid(conn) do
+    case GenServer.call(conn, {:set_option, to_string(key), value}) do
+      :ok ->
+        :ok
+
+      {:error, reason} ->
+        {:error, error_to_exception(reason)}
+    end
+  end
+
+  @doc """
+  Set a string type option for the connection.
+  """
+  @spec set_string_option(pid(), atom() | String.t(), term()) ::
+          :ok | {:error, String.t()}
+  def set_string_option(conn, key, value) when is_pid(conn) do
+    case GenServer.call(conn, {:set_option, to_string(key), to_string(value)}) do
```

Comment:
I think we can simplify this as:

    GenServer.call({:set, :adbc_connection_set_option_string, [to_string(key), to_string(value)]})

And then in the GenServer you call the NIF

    def handle_call({:set, fun, args}, _from, state) do
      case apply(Adbc.NIF, fun, [state.ref | args]) do
        ...
      end
    end

This way you can skip all of the helpers in adbc_helper? It would also be nice to make the NIF name consistent too: `adbc_connection_set_string_option` and so forth. :)